src/Analyzers/Core/CodeFixes/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsCodeFixProvider.cs

```diff
@@ -7,20 +7,25 @@
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
 using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
 
-internal abstract class AbstractRemoveUnnecessaryImportsCodeFixProvider : CodeFixProvider
+internal abstract class AbstractRemoveUnnecessaryImportsCodeFixProvider : SyntaxEditorBasedCodeFixProvider
 {
+    protected abstract string GetTitle();
     protected abstract ISyntaxFormatting GetSyntaxFormatting();
 
     public sealed override ImmutableArray<string> FixableDiagnosticIds
         => [RemoveUnnecessaryImportsConstants.DiagnosticFixableId];
 
-    public sealed override FixAllProvider GetFixAllProvider()
-        => WellKnownFixAllProviders.BatchFixer;
+#if !CODE_STYLE
+    // We don't need to perform any cleanup for this code fix.  It simply removes code, leaving
+    // things in the state it already needs to be in.
+    protected override CodeActionCleanup Cleanup => CodeActionCleanup.None;
+#endif
```

Comment:
this ensures that after the fixer runs, the Code Action system itself doesn't do an full semantic pass on the resultant documents to clean them up.